File: 프로그래머스/lv2/42839. 소수 찾기/소수 찾기.js
const isPrime = (n) => {
    if (n <= 1) return false;
    for (let i = 2; i <= Math.sqrt(n); ++i)
        if (n % i === 0) return false;
    return true;
}

function solution(numbers) {
    var answer = 0;
    const arr = numbers.split('');
    const check = {}, visit = {};

    (function dfs(value) {
        if (!check[value] && isPrime(value)) ++answer;
        check[value] = true;

        for (let i = 0; i < arr.length; ++i) {
            if (visit[i]) continue;

            visit[i] = true;
            dfs((value.toString() + arr[i]) * 1, i);
            visit[i] = false;
        }
    })(0);

    return answer;
}

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.cpp
#include <string>
#include <vector>
#include <unordered_set>
#include <algorithm>
#include <cmath>
#include <iostream>

using namespace std;

const bool IsPrime(int n)
{
    for (size_t i = 2; i <= sqrt(n); ++i)
        if (n % i == 0) return false;
    return true && n > 1;
}

void expression(const string& numbers, unordered_set<int>& set, vector<int> idxList, string num)
{
    set.insert(stoi(num));

    for (int i = 0; i < numbers.length(); ++i)
    {
        if (find(idxList.begin(), idxList.end(), i) != idxList.end()) continue;

        idxList.push_back(i); num += numbers[i];
        expression(numbers, set, idxList, num);
        idxList.pop_back(); num.pop_back();
    }
}

int solution(string numbers) {
    //24.10.21
    int answer = 0;
    unordered_set<int> set;
    
    for (int i = 0; i < numbers.length(); ++i)
    {
        vector<int> idxList{i};
        string num;
        num += numbers[i];
        expression(numbers, set, idxList, num);
    }

    for (const auto& n : set)
        answer += IsPrime(n) ? 1 : 0;

    return answer;
}